projects/laji/src/app/shared/service/project-form.service.ts

```diff
@@ -31,16 +31,48 @@ export interface NamedPlacesRouteData extends NamedPlacesQueryModel {
   namedPlace?: NamedPlace;
 }
 
-@Injectable()
+@Injectable({providedIn: 'root'})
 export class ProjectFormService {
   constructor(
     private formService: FormService,
     private translate: TranslateService,
     private namedPlacesService: NamedPlacesService
   ) { }
 
-  getFormFromRoute$(route: ActivatedRoute) {
-    return this.getFormID(route).pipe(switchMap(formID => this.formService.getForm(formID, this.translate.currentLang)));
+  private currentFormID: string;
+  private form$: ReplaySubject<Form.SchemaForm>;
+
+  /**
+   * LajiFormBuilder can change the language of the form, without changing the lang of the whole page.
+   */
+  public localLang$ = new BehaviorSubject<string>(this.translate.currentLang);
+  public remountLajiForm$ = new Subject<void>();
+
+  getFormFromRoute$(route: ActivatedRoute): Observable<Form.SchemaForm> {
+    return this.getFormID(route).pipe(switchMap(formID => this.getForm(formID)));
+  }
+
+  getForm(id: string): Observable<Form.SchemaForm> {
```

Comment:
The naming could be more consistent, some have `$` in their name (`getFormFromRoute$`) and some don't (`getForm`)